Sources/NIO/Interfaces.swift

```diff
@@ -462,6 +462,11 @@ public final class NIONetworkInterface {
 #else
     internal init?(_ caddr: ifaddrs) {
         self.name = String(cString: caddr.ifa_name)
+
+        guard caddr.ifa_addr != nil else {
+            return nil
```

Comment:
Android devices have a bunch of interfaces, many without an address, which was breaking the next `guard` when this is called from `enumerateInterfaces()`. Since that function is deprecated anyway, could just ignore it and remove this change.

Reply:
Yeah, that's part of why this was deprecated. Nonetheless, a crash is a bad idea and I'm fine with this change.